client/src/sections/resources/Banishers.tsx

```diff
@@ -0,0 +1,171 @@
+import React from "react";
+import Line from "../../components/Line";
+import Tile from "../../components/Tile";
+import { useHaveEquipped, useMyTurncount } from "../../hooks/useCall";
+import useGet from "../../hooks/useGet";
+import useHave from "../../hooks/useHave";
+import { returnBanishes } from "../../util/chunkBanish";
+import { $item, $skill } from "../../util/makeValue";
+import { plural } from "../../util/text";
+
+/**
+ * Class used to store banishers. Unless otherwise specified, all params are strings.
+ * @param name The name of the banisher source
+ * @param numberRemaining Number of the banishers remaining (#)
+ * @param access Boolean representing if the user can access the banish
+ * @param banishType Must be "skill", "item", or "equip"; kind of banish
+ * @param href (optional) The target for a link
+ * @param isEquipped (optional) Boolean of if the item needed is equipped
+ * @param equipItem (optional) String of the item you need to equip
+ * @param extraString (optional) An extra string to append to the end of the tile
+ */
+class BanisherObject {
+  name: string;
+  numberRemaining: number;
+  access: boolean;
+  banishType: string;
+  href: string;
+  banishedMonster: string;
+  isEquipped: boolean;
+  equipItem: string;
+  extraString: string;
+
+  constructor(
+    name: string,
+    numberRemaining: number,
+    access: boolean,
+    banishType: "skill" | "item" | "equip",
+    href?: string,
+    banishedMonster?: string,
+    isEquipped?: boolean,
+    equipItem?: string,
+    extraString?: string
+  ) {
+    this.name = name;
+    this.numberRemaining = numberRemaining;
+    this.access = access;
+    this.banishType = banishType;
+    this.banishedMonster = banishedMonster ?? "";
+    this.href = href ?? "";
+    this.isEquipped = isEquipped ?? true;
+    this.equipItem = equipItem ?? "";
+    this.extraString = extraString ?? "";
+  }
+
+  // Returns if the banish is accessible for filtering.
+  accessible(): boolean {
+    return this.access && this.numberRemaining > 0;
+  }
+
+  // Renders the tile. Switches the line's stylings based on banishType.
+  render(): React.ReactNode {
+    const possibleStrings: { [key: string]: string } = {
+      skill: `${plural(this.numberRemaining, "cast")} of ${this.name}`,
+      item: `${plural(this.numberRemaining, "use")} of ${this.name}`,
+      equip: `${plural(this.numberRemaining, "use")} of ${this.name}${
+        !this.isEquipped ? ` (equip your ${this.equipItem}!)` : ""
+      }`,
+    };
+
+    return <Line href={this.href}>{possibleStrings[this.banishType]}</Line>;
+  }
+}
+
```

Comment:
This feels like a very heavyweight approach for a component that is really just storing data.

Reply:
i agree that this seems a bit chunky-for-purpose on the outside. but i really think this heavyweight approach is superior to the minimalist approach currently used in freeFights.tsx. imo: 

- this makes it much easier for future folks to add new banishes
- this makes it much easier to change the output format of the rendered banish lines without having to modify all the elements individually (which will be especially useful if i make a version of the chevron code that can be used for, say, end-of-line progress bars or something, or we add more equipLink support)
- this keeps the underlying code a bit better organized and easier to read 

i definitely agree that there is room to lighten the number of fields in the class, for sure... but i think overall the bones of this slighty heavy class based approach are pretty solid here. i think i would need to see an example of a better lightweight approach to this tile this to be convinced otherwise tbh.

Reply:
Why not just make it a plain old object, and then pass that object to a new component that renders it? I would suggest that whenever you are making a class that returns a ReactNode you are probably supposed to be making a component somehow.

Reply:
isEquipped being a property rather than just calling the function feels bad, but also I guess the way hooks work you can't do it any other way :-/

Reply:
I feel like in general this would benefit a lot from being able to access mafia functions more directly if possible--equipItem probably should be something that this can just Do, rather than relying on it being an input.. Bean is right insofar as this seems to be adding a lot of structure, and not benefiting from it. I know there have been some major shifts since last I worked on YORICK, and I'm curious if there's anything you can do here that actually helps. Otherwise, this entire class feels mostly like it could be a single function that takes all the inputs of the constructor and does the rendering itself.